### PR TITLE
Knex migration

### PR DESCRIPTION
**Pull request by ColorfulCodes**
Here is a sample Knex migration to create a links table.

- [x] Add an `env` sample file
- [x] Document setup instructions in `README.md`

### CODE REVIEW

File: migrations/20190606092823_create_project_links_table.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+exports.up = knex => (
+  knex.schema
+    .createTable('project_links', (table) => {
+      table.increments('id');
+      table.string('project_id', 255);
+      table.string('lable', 255);
```
Comment (cromwellryan) [inline migrations/20190606092823_create_project_links_table.js]: Misspelled `label` as `lable`

File: migrations/makeTable.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+exports.up = function(knex) {
+  return knex.schema
+    .createTable('user', function (table) {
+       table.increments('id');
```
Comment (cromwellryan) [inline migrations/makeTable.js]: I don't think we need this :)

Comment (ColorfulCodes) [reply migrations/makeTable.js]: Deleted.

File: migrations/20190606092823_create_project_links_table.js (hunk 1 above)
Comment (ColorfulCodes) [reply migrations/20190606092823_create_project_links_table.js]: Fixed.

File: knexfile.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+var env = require('node-env-file');
+env(__dirname + '/.env');
+
+module.exports = {
+
+  development: {
+    client: 'mysql',
+    connection: {
+      database: process.env.dbName,
```
Comment (cromwellryan) [inline knexfile.js]: All env variables should be capitalized. So here it would be `DBNAME`, for example.

Comment (ColorfulCodes) [reply knexfile.js]: resolved.

File: envFileExample.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#This is an example of your .env file within your Slackbot project.
+
+# SlackBot Environment Configuration which can be found in your api.slack.com account
+clientId=
+clientSecret=
+clientSigningSecret=
+
+# Here you will include your Knex File Configuration Information
+dbName=NameofDatabase
```
Comment (cromwellryan) [inline envFileExample.txt]: These don't line up with the `DBNAME` properties used in the `knexfile.js`.

File: readme.md [hunk 5]
```diff
@@ -12,10 +12,18 @@ delete specific information. See the image below to get a better understanding.
 ## End Goal
 ![Project Bot End Result](Images/slash.png)
 
+
+
+### Knex Configuration:
+In order for your Knex migrations to function correctly be sure to setup your
+local project .env file with your own personal host, user, password and database name
```
Comment (cromwellryan) [inline readme.md]: Maybe put `.env` in back ticks (`). Same with the example and .gitignore.

File: envFileExample.txt (hunk 4 above)
Comment (ColorfulCodes) [reply envFileExample.txt]: Will do.

File: readme.md (hunk 5 above)
Comment (ColorfulCodes) [reply readme.md]: Done.

**Pull request merged**